internal/common/frontendo11yapi/client.go

```diff
@@ -0,0 +1,233 @@
+	path := fmt.Sprintf("%s/app", pathPrefix)
+	var apps []App
+	err := c.doAPIRequest(ctx, baseURL, stackID, http.MethodGet, path, nil, &apps)
+	if err != nil {
+		return []App{}, fmt.Errorf("failed to get faro apps: %w", err)
+	}
+	return apps, nil
+}
+
+func (c *Client) GetApp(ctx context.Context, baseURL string, stackID int64, appID int64) (App, error) {
+	path := fmt.Sprintf("%s/app/%d", pathPrefix, appID)
+	var app App
+	err := c.doAPIRequest(ctx, baseURL, stackID, http.MethodGet, path, nil, &app)
+	if err != nil {
+		return App{}, fmt.Errorf("failed to get faro apps: %w", err)
+	}
+	return app, nil
+}
+
+func (c *Client) UpdateApp(ctx context.Context, baseURL string, stackID int64, appID int64, appData App) (App, error) {
+	path := fmt.Sprintf("%s/app/%d", pathPrefix, appID)
+	var app App
+	err := c.doAPIRequest(ctx, baseURL, stackID, http.MethodPut, path, &appData, &app)
+	if err != nil {
+		return App{}, fmt.Errorf("failed to update faro app %q: %w", appData.Name, err)
+	}
+	return app, nil
+}
+
+func (c *Client) DeleteApp(ctx context.Context, baseURL string, stackID int64, appID int64) error {
+	path := fmt.Sprintf("%s/app/%d", pathPrefix, appID)
+	err := c.doAPIRequest(ctx, baseURL, stackID, http.MethodDelete, path, nil, nil)
+	if err != nil {
+		return fmt.Errorf("failed to delete faro app id=%d: %w", appID, err)
+	}
+	return nil
+}
+
+var (
+	ErrNotFound     = fmt.Errorf("not found")
+	ErrUnauthorized = fmt.Errorf("request not authorized for stack")
+)
+
+func (c *Client) doAPIRequest(ctx context.Context, rawURL string, stackID int64, method string, path string, body any, responseData any) error {
+	parsedURL, err := url.Parse(rawURL)
+	if err != nil {
+		return fmt.Errorf("failed to parse frontend o11y API url: %w", err)
+	}
+
+	var reqBodyBytes io.Reader
+	if body != nil {
+		bs, err := json.Marshal(body)
+		if err != nil {
+			return fmt.Errorf("failed to marshal request body: %w", err)
+		}
+		reqBodyBytes = bytes.NewReader(bs)
+	}
+
+	req, err := http.NewRequestWithContext(ctx, method, parsedURL.String()+path, reqBodyBytes)
+	if err != nil {
+		return fmt.Errorf("failed to create request: %w", err)
+	}
+
+	for k, v := range c.defaultHeaders {
+		req.Header.Add(k, v)
+	}
+
+	req.Header.Add("Authorization", fmt.Sprintf("Bearer %d:%s", stackID, c.authToken))
+	req.Header.Add("X-Scope-OrgID", fmt.Sprintf("%d", stackID))
+	req.Header.Add("Content-Type", "application/json")
+	req.Header.Add("User-Agent", c.userAgent)
+
+	resp, err := c.client.Do(req)
+	if err != nil {
+		return fmt.Errorf("failed to do request: %w", err)
+	}
+
+	bodyContents, err := io.ReadAll(resp.Body)
+	resp.Body.Close()
+	if err != nil {
+		return fmt.Errorf("failed to read response body: %w", err)
+	}
+	if !(resp.StatusCode >= 200 && resp.StatusCode <= 299) {
+		if resp.StatusCode == 404 {
+			return ErrNotFound
+		}
+		if resp.StatusCode == 401 {
+			return ErrUnauthorized
+		}
+		return fmt.Errorf("status: %d", resp.StatusCode)
+	}
```

Comment:
Its a bit complex to read, maybe this?

```suggestion
	if resp.StatusCode < http.StatusOK || resp.StatusCode >= http.StatusMultipleChoices {
	        switch resp.StatusCode {
		case http.StatusNotFound:
			return ErrNotFound
		case http.StatusUnauthorized:
			return ErrUnauthorized
		default:
			return fmt.Errorf("status: %d", resp.StatusCode)
		}
	}
```

Reply:
I am personally in favour of using hardcoded values to indicate that an error is within the 2xx space as I don't remember by hand that Multiple Choices is 300. Other than that I will switch this to a switch, looks cleaner indeed